### DISCUSSION TITLE
Support for v3.1 Specs

### DISCUSSION THREAD

**Discussion by AlaNDGT** [OpenAPITools/openapi-diff#345] [New Features]
Hi,

Are there plan to support version 3.1? 

Currently i get the following error:

```
Unexpected exception. Reason: Cannot read old OpenAPI spec
java.lang.RuntimeException: Cannot read old OpenAPI spec
	at org.openapitools.openapidiff.core.OpenApiCompare.notNull(OpenApiCompare.java:107)
	at org.openapitools.openapidiff.core.OpenApiCompare.fromSpecifications(OpenApiCompare.java:102)
	at org.openapitools.openapidiff.core.OpenApiCompare.fromLocations(OpenApiCompare.java:91)
	at org.openapitools.openapidiff.cli.Main.main(Main.java:174)
```

**Comment by joschi** [comment]
> Are there plan to support version 3.1?

We're planning to support OpenAPI 3.1 as soon as the upstream projects swagger-core and swagger-parser support this version of the specification.

You can follow the development at https://github.com/swagger-api/swagger-parser/issues/1535.

**Reply by phillipuniverse** [reply]
@joschi seems like the latest version of swagger-parser works with OpenAPI 3.1. What's the latest status on OpenAPI 3.1 support in this tool?

**Reply by joschi** [reply]
@phillipuniverse Did you try the latest (beta) release with an OpenAPI 3.1 spec? 😬
https://github.com/OpenAPITools/openapi-diff/releases/tag/2.1.0-beta.7

**Reply by oleg-kovalov** [reply]
@joschi could you please elaborate:
- with swagger-core and swagger-parser now both support OAS 3.1 is it safe to say openapi-diff supports it as well? Even thought the latest release is beta. 

Thanks!

**Reply by joschi** [reply]
@oleg-kovalov Yes, it should be supported.

It would be great if you could try it out with the latest beta release and let us know the result. 😉
